lib/services/local/presence_test.go

```diff
@@ -687,6 +687,40 @@ func TestListResources(t *testing.T) {
 			}, time.Second, 100*time.Millisecond)
 			require.Empty(t, nextKey)
 
+			// list resources only with matching search keywords
```

Comment:
Looks like you were just following the pattern already used in this test, but I think this might be a good opportunity to make this test better. What do you think about the following?

1. Run the tests in parallel. In order to do this I think you will need to move the creation of `backend` and `presence` inside `t.Run` and change the test struct to the following
`createResourceFunc     func(context.Context, *PresenceService, string, map[string]string) error
deleteAllResourcesFunc func(context.Context, *PresenceService) error`

2. We probably don't need to store all the resources in `resultResourcesXXXXXX`. Since `require.Eventually` is only checking that the length matches the expected number, we could change `resultResourcesXXXXXX` to be an int and replace all the `resultResourcesXXXXXX = append(....)` with `resultResourcesXXXXXX += len(resources)`